### ISSUE TITLE
universe: add background loop for optimistic asset proof to FederationEnvoy

### ISSUE THREAD

**Issue by Roasbeef** [lightninglabs/taproot-assets#526]
Today after we create an asset, if we have an active universe federation, we'll attempt to push out the new proofs to them. However, if this fails for w/e reason, then we won't attempt to every push out proofs again, unless someone connects to our Universe server and then obtains our proofs, and pushes out to others. 

We should add a background reconciliation loop, that if configured, will push out our proofs to our universe federation. A simple version of this would just be to grab our issuance proofs (or full, etc) then attempt to call an push out to our federation members. This would be somewhat inefficient. Another take would be to instead maintain an on disk queue, and work through the queue ensuring that each time the proof is accepted. If so, then we can stop trying to publish it. 

## Steps To Completion

- [x] Decide in if we'll do best effort basis (not efficient), or use an on disk queue, and treat pushing to a universe like we do sending proof files after a transfer. 
- [ ] Implement option above, figure out how to expose as configuration

**Comment by Roasbeef** [comment]
This TODO can be removed once this is fixed: https://github.com/lightninglabs/taproot-assets/blob/a201d48152b274e8faec390dab969391171af27e/universe/auto_syncer.go#L274-L276

**Comment by dstadulis** [comment]
Is the background loop currently pushing? may be only receiving

**Comment by dstadulis** [comment]
This will ensure that the proof is more properly distributed during the minting process because there isn't a second attempt on the proof publishing.

**Comment by ffranr** [comment]
> maintain an on disk queue, and work through the queue ensuring that each time the proof is accepted. If so, then we can stop trying to publish it.

I'm in favour of this approach.

I think we only need to care about syncing our issuance proofs. And not transfer proofs.

Perhaps a table with columns:
```
[issuance_proof, universe_server, timestamp, proof_sync_status]
```

Where `proof_sync_status` is an enum with values `unknown`, `accepted`, and `rejected`.

This will hopefully give us enough flexibility to see our latest sync history. And also to force re-try based on the sync timestamp, the sync status, and specific to a particular universe server.

Configuration wise, we already have a default sync interval of 10 mins. I think that config parameter can be re-used for this new case. When combined with our issuance proof sync history SQL table we shouldn't spam other universe servers excessively.

One bit of work that we might be missing is the response from the other universe servers. It would be useful if a peer universe server could include in its response whether the proof was rejected (server not interested in proof) or accepted (no need to try again, proof has been recognised).

**Comment by jharveyb** [comment]
Agree re: issuance only.

I think this could look similar to what we do for tracking asset transfers? And that what you've outlined sounds similar to that IIUC.

So we can make the initial write with status `pending`, then update to `sent`.

I looked around the call path for this upload and IIUC a universe server would return an error if proof validation fails, as well as some intermediate step of parsing the universe ID. That remote server would also already return an error about rejecting proof insertion for that universe ID. So I don't think changes are needed there re: errors being propagated to the minter. And if a proof is accepted, the `nil` err should be propagated to the minter.

IMO the goal is to make sure proof upload was _attempted_ successfully; if a universe in my federation decides to reject all new issuance proofs, I shouldn't retry that upload forever. I only want to ensure that I connected to them and actually sent the proof.

Concretely, i think this on-disk queue would mirror the `pushRequests` and `batchPushRequests` channels in `FederationEnvoy`:

https://github.com/lightninglabs/taproot-assets/blob/12d24398b8027b592fbf51d0ed17f4f21a3d30c0/universe/auto_syncer.go#L342

So once we successfully register a proof with our local registrar, add it to the on-disk queue. Then have another case stmt here to handle queue reading and rebroadcast, etc.

IMO we could directly use the backoff retry logic we already use for proof transfer during send/recv.
